### ISSUE TITLE
Build fails after upgrading tls / x509 / nocrypto

### ISSUE THREAD

**Issue by verbosemode** [hannesm/tlstunnel#26]
I just ran my daily opam upgrade and the tlstunnel build failed. It seems like Tls.Printer doesn't exist anymore.

Here is the full text output:

```
~  ᐅ ouu

=-=- Updating package repositories =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[xmpp-dev] git://github.com/hannesm/xmpp-opam updated
[mirage-dev] git://github.com/mirage/mirage-dev already up-to-date
[default] synchronized from https://opam.ocaml.org

=-=- Synchronizing development packages -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[BetterErrors] https://github.com/chenglou/ocaml-better-errors.git already up-to-date
[erm_xmpp] https://github.com/hannesm/xmpp.git#otr updated
[jackline] https://github.com/hannesm/jackline.git updated
[WARNING] A conflict was detected in your installation. This can be caused by updated constraints or conflicts in your installed packages:
            - Conflicting version constraints for nocrypto

The following dependencies are in cause:
  - jackline -> erm_xmpp >= 0.3 -> nocrypto >= 0.5.3

You should run "opam upgrade" to resolve the situation.
The following actions will be performed:
  ↗  upgrade   nocrypto       0.5.2 to 0.5.3
  ↗  upgrade   x509           0.5.0 to 0.5.1
  ↗  upgrade   otr            0.3.0 to 0.3.1
  ↻  recompile erm_xmpp       0.3             [uses nocrypto]
  ↗  upgrade   tls            0.7.0 to 0.7.1
  ↻  recompile tlstunnel      0.1.2           [uses tls, nocrypto, x509]
  ↻  recompile jackline       0.1.0           [uses tls, nocrypto, otr, x509]
  ↻  recompile conduit        0.10.0          [uses tls]
  ↻  recompile mirage-conduit 2.2.0           [uses tls]
  ↻  recompile cohttp         0.19.3          [uses conduit]
  ↻  recompile podge          0.5             [uses cohttp]
  ↻  recompile mirage-http    2.5.1           [uses cohttp, mirage-conduit]
  ↻  recompile cowabloga      0.2.0           [uses cohttp]
  ↻  recompile c3             0.3.0           [uses cohttp]
  ↻  recompile git            1.7.2           [uses nocrypto]
  ↻  recompile mirage-git     1.6.0           [uses git]
  ↻  recompile git-unix       1.7.1           [uses nocrypto]
  ↻  recompile irmin          0.10.1          [uses git]
  ↻  recompile mirage-irmin   0.9.8           [uses mirage-git]
  ↻  recompile irmin-unix     0.9.9           [uses git-unix]
===== ↻  16   ↗  4 =====
Do you want to continue ? [Y/n] Y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[cohttp] Archive in cache
[conduit] Archive in cache
[cowabloga] Archive in cache
[git] Archive in cache
[irmin] Archive in cache
[erm_xmpp] https://github.com/hannesm/xmpp.git#otr already up-to-date
[mirage-http] Archive in cache
[jackline] https://github.com/hannesm/jackline.git already up-to-date
[default] https://opam.ocaml.org/archives/c3.0.3.0+opam.tar.gz downloaded
[podge] Archive in cache
[default] https://opam.ocaml.org/archives/otr.0.3.1+opam.tar.gz downloaded
[tlstunnel] Archive in cache
[default] https://opam.ocaml.org/archives/x509.0.5.1+opam.tar.gz downloaded
[default] https://opam.ocaml.org/archives/tls.0.7.1+opam.tar.gz downloaded
[default] https://opam.ocaml.org/archives/nocrypto.0.5.3+opam.tar.gz downloaded

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
⊘  removed   c3.0.3.0
⊘  removed   cowabloga.0.2.0
⊘  removed   irmin-unix.0.9.9
⊘  removed   jackline.0.1.0
⊘  removed   erm_xmpp.0.3
⊘  removed   mirage-irmin.0.9.8
⊘  removed   irmin.0.10.1
⊘  removed   git-unix.1.7.1
⊘  removed   mirage-git.1.6.0
⊘  removed   git.1.7.2
⊘  removed   mirage-http.2.5.1
⊘  removed   mirage-conduit.2.2.0
⊘  removed   otr.0.3.0
⊘  removed   podge.0.5
⊘  removed   cohttp.0.19.3
⊘  removed   conduit.0.10.0
⊘  removed   tlstunnel.0.1.2
⊘  removed   tls.0.7.0
⊘  removed   x509.0.5.0
⊘  removed   nocrypto.0.5.2
∗  installed nocrypto.0.5.3
∗  installed erm_xmpp.0.3
∗  installed otr.0.3.1
∗  installed x509.0.5.1
∗  installed tls.0.7.1
∗  installed conduit.0.10.0
∗  installed cohttp.0.19.3
∗  installed c3.0.3.0
∗  installed cowabloga.0.2.0
∗  installed jackline.0.1.0
∗  installed mirage-conduit.2.2.0
∗  installed mirage-http.2.5.1
∗  installed git.1.7.2
∗  installed git-unix.1.7.1
∗  installed mirage-git.1.6.0
∗  installed irmin.0.10.1
∗  installed irmin-unix.0.9.9
∗  installed mirage-irmin.0.9.8
∗  installed podge.0.5
[ERROR] The compilation of tlstunnel failed at "ocaml pkg/build.ml native=true native-dynlink=true".

#=== ERROR while installing tlstunnel.0.1.2 ===================================#
# opam-version 1.2.2
# os           linux
# command      ocaml pkg/build.ml native=true native-dynlink=true
# path         /home/lobo/.opam/4.02.3/build/tlstunnel.0.1.2
# compiler     4.02.3
# exit-code    10
# env-file     /home/lobo/.opam/4.02.3/build/tlstunnel.0.1.2/tlstunnel-5393-e01822.env
# stdout-file  /home/lobo/.opam/4.02.3/build/tlstunnel.0.1.2/tlstunnel-5393-e01822.out
# stderr-file  /home/lobo/.opam/4.02.3/build/tlstunnel.0.1.2/tlstunnel-5393-e01822.err
### stdout ###
# ocamlfind ocamldep -package tls -package x509 -package nocrypto -package lwt -package cmdliner -package sexplib -package lwt.unix -package tls.lwt -modules tlstunnel.ml > tlstunnel.ml.depends
# ocamlfind ocamlc -c -annot -bin-annot -package tls -package x509 -package nocrypto -package lwt -package cmdliner -package sexplib -package lwt.unix -package tls.lwt -w +A-4-6-7-9-40-42-44-48 -warn-error +1..49 -o tlstunnel.cmo tlstunnel.ml
# + ocamlfind ocamlc -c -annot -bin-annot -package tls -package x509 -package nocrypto -package lwt -package cmdliner -package sexplib -package lwt.unix -package tls.lwt -w +A-4-6-7-9-40-42-44-48 -warn-error +1..49 -o tlstunnel.cmo tlstunnel.ml
# File "tlstunnel.ml", line 169, characters 16-49:
# Error: Unbound module Tls.Printer
# Hint: Did you mean Writer?
# Command exited with code 2.



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install tlstunnel 0.1.2
The following changes have been performed
  ⊘  remove  tlstunnel 0.1.2         
  ↗  upgrade nocrypto  0.5.2 to 0.5.3
  ↗  upgrade otr       0.3.0 to 0.3.1
  ↗  upgrade tls       0.7.0 to 0.7.1
  ↗  upgrade x509      0.5.0 to 0.5.1

The former state can be restored with:
    opam switch import "~/.opam/4.02.3/backup/state-20160222122642.export"
```

**Comment by hannesm** [comment]
thx, adjusted the constraints in opam-repository https://github.com/ocaml/opam-repository/pull/6089 , fixed in commit e6c0eec (will do a release)

**Comment by hannesm** [comment]
(release 0.1.3 https://github.com/ocaml/opam-repository/pull/6090)
